test/indexTest.js

```diff
@@ -152,3 +152,102 @@ test('key derivation', (t) => {
   t.deepEqual(nacl.sign.open(signed, key.publicKey), null, 'verification failure')
   t.throws(crypto.deriveSigningKeysFromSeed.bind(null, []), /Uint8Array/, 'error when input is not a Uint8Array')
 })
+
+const pair = nacl.sign.keyPair.fromSecretKey(
+  Uint8Array.from(
+    Buffer.from('9f8362f87a484a954e6e740c5b4c0e84229139a20aa8ab56ff66586f6a7d29c526b40b8f93fff3d897112f7ebc582b232dbd72517d082fe83cfb30ddce43d1bb', 'hex')
+  )
+)
+const goodSignature = 'keyId="test-key-ed25519",algorithm="ed25519",headers="foo fizz",signature="lAGT9Bhde3sJp8Z1NTxmViJtG1PSoYnXV9he82z1iu//KXmCrjKYe1JOU34memKIdlxG1yJoeS2hxANRvalrBw=="'
+
+test('signing', (t) => {
+  const headers = { foo: 'bar', fizz: 'buzz' }
+  t.plan(6)
+  
+  let signature = crypto.ed25519HttpSign('test-key-ed25519', pair.secretKey, headers);
```

Comment:
it bothers me that some lines have semicolons and some don't, but we can lint the test files in a follow-up. tests lgtm

Reply:
Hard habit to break 😅